smartapps/curb/curb-energy-manager.src/curb-energy-manager.groovy

```diff
@@ -0,0 +1,248 @@
+      input("meter", "capability.energyMeter", title: "Select Monitored Circuit", multiple: false)
+      input("circuits", "capability.energyMeter", title: "Circuits to send alerts on", multiple:true, defaultValue: ALL)
+    }
+  }
+    page(name: "pageThree", install: true, uninstall:true) {
+    section("Controlled Appliances") {
+      input("thermostats", "capability.thermostat", title: "Select your Thermostat", multiple: true, required: false)
+      input("switches", "capability.switch", title: "Select your Load Controllers", multiple: true, required: false)
+            //input("cycleTimeLatency", "float", title: "Set Cycle Time Latency (minutes)", defaultValue: 5)
+    }
+
+    section("Notifications"){
+          input("recipients", "contact", title: "Send notifications to") {
+              input "phone", "phone", title: "Phone Number",
+                  description: "Recieve Text Messages about your usage", required: false
+          }
+      }
+  }
+}
+
+def installed() {
+  resetClocking()
+  //log.debug "Installed with settings: ${settings}"
+
+  initialize()
+}
+
+def updated() {
+  //log.debug "Updated with settings: ${settings}"
+  runAutomation()
+  unsubscribe()
+  initialize()
+}
+
+def initialize() {
+  subscribe(meter, "power", checkEnergyMonitor)
+  runEvery1Minute(runAutomation)
+
+}
+
+def checkRunning() {
+  def df = new java.text.SimpleDateFormat("EEEE")
+  df.setTimeZone(location.timeZone)
+  if (weekdays.contains(df.format(new Date()))) {
+    def hf = new java.text.SimpleDateFormat("H")
+    hf.setTimeZone(location.timeZone)
+    if (hours.contains(hf.format(new Date()).toString())) {
+      return true
+    }
+  }
+  return false
+}
+
+def sendNotifications() {
+  //log.debug("notifying")
+    //log.debug(recipients)
+    def devlist = []
+    //log.debug(circuits)
+    def count = 0
+    def currentTotal = Float.parseFloat(meter.currentState("power").value)
+    for(c in circuits){
+        //log.debug(c)
+        //log.debug(currentTotal)
+        try {
+        devlist.add([pct: ((Float.parseFloat(c.currentState("power").value) / currentTotal) * 100).round(), name: c.toString()])
+        count += count
+        } catch (e) {
+          log.debug(e);
+        }
+
+    }
+
+    //log.debug(devlist)
+
+    def sorted = devlist.sort { a, b -> b.pct <=> a.pct }
+    def message = "Curb Alert: Energy usage is projected to go over selected threshold."
+    //log.debug(devlist.size())
+    if (devlist.size() > 3)
+    {
+      message = "Curb Alert: Energy usage is projected to go over selected threshold. Your biggest consumers currently are: ${sorted[0].name} ${sorted[0].pct}%, ${sorted[1].name} ${sorted[1].pct}%, and ${sorted[2].name} ${sorted[2].pct}%"
+    }
+  sendNotificationToContacts(message, recipients)
+}
+
+def resetClocking() {
+  //log.debug("resetting the clock")
+  state.readings = []
+  for (int i = 0; i < Integer.parseInt(timeInterval); i++) {
```

Comment:
Is this null assignment really necessary?
The array is already cleared on ln129, accessing any element in the array should result in a null response unless they've been assigned to something else.